http-server-netty/src/main/java/io/micronaut/http/server/netty/NettyHttpServer.java

```diff
@@ -657,4 +604,221 @@ public ChannelGroup getChannelGroup() {
     public WebSocketSessionRepository getWebSocketSessionRepository() {
         return this;
     }
+
+    private HttpToHttp2ConnectionHandler newHttpToHttp2ConnectionHandler() {
+        Http2Connection connection = new DefaultHttp2Connection(true);
+        final InboundHttp2ToHttpAdapter http2ToHttpAdapter = new InboundHttp2ToHttpAdapterBuilder(connection)
+                .validateHttpHeaders(true)
```

Comment:
Don't you need to use `serverConfiguration.isValidateHeaders()` here?

Reply:
Good catch